src/components/modal/index.tsx

```diff
@@ -32,6 +36,32 @@ export interface ModalProps {
 }
 
 export const Modal: React.FC<ModalProps> = ({ title, show, onExit, children }) => {
+    const ref = useRef<HTMLDivElement>(null);
+    // Close modal on Esc press
+    const handleEscKey = useCallback((event: KeyboardEvent) => {
+        if (event.key === KEY_NAME_ESC) {
+            onExit();
+        }
+    }, [onExit]);
+
+    // Close on outside click
+    const handleClick = useCallback((event: MouseEvent) => {
+        // @ts-ignore EventTarget is not mapping on Node but works just fine
+        if (ref?.current?.contains && !ref.current.contains(event.target)) {
+            onExit();
+        }
+    }, [onExit, ref]);
```

Comment:
Or we can just bind the `onClick` action to the background div instead.

Reply:
Just noticed a nasty bug that ANY click closes the modal (even on the text field). Reverting the ref solution.

Reply:
Oh yeah I think you need `e.stopPropagation()`. I can push a fix for that.

Reply:
I tried that and it didn't work, because there are no explicit handlers on the text fields etc.